File: 0003/LongestSubstringWithoutRepeatingCharacters.c
int lengthOfLongestSubstring(char* s) {
    int index[256] = {0};
    int first_index = -1;
    int length = strlen(s);
    int result = 0;

    for(int i = 0; i < length; ++i){
        first_index = index[s[i]] > first_index ? index[s[i]] : first_index;
        index[s[i]] = i + 1;
        result = (i - first_index + 1) > result ? (i - first_index + 1) : result;
    }

    return result;
}

File: 0003/LongestSubstringWithoutRepeatingCharacters.swift
class Solution {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        var longestLength = 0, startIndex = 0
        var charMap: [Character: Int] = [:]

        for (index, char) in s.enumerated() {
            if let foundIndex = charMap[char] {
                startIndex = max(foundIndex + 1, startIndex)
            }
            longestLength = max(longestLength, index - startIndex + 1)
            charMap[char] = index
        }

        return longestLength
    }
}